src/parser/grammar.ne

```diff
@@ -202,7 +208,7 @@ atomic_expression ->
 array_subscript -> %ARRAY_IDENTIFIER _ square_brackets {%
   ([arrayToken, _, brackets]) => ({
     type: NodeType.array_subscript,
-    array: addComments({ type: NodeType.identifier, text: arrayToken.text}, { trailing: _ }),
+    array: addComments({ type: NodeType.identifier, tokenType: TokenType.ARRAY_IDENTIFIER, text: arrayToken.text}, { trailing: _ }),
```

Comment:
So, currently `ARRAY_IDENTIFIER` tokens are treated differently from normal identifiers. This results in the following SQL:

```sql
select foo, foo[1] from tbl
```

being formatted as:

```sql
select
  FOO,
  foo[1]
from
  TBL
```

That is, when the `foo` column is used normally, it gets uppercased, but when used together with array-accessor operator it is not uppercased.

This difference between normal identifiers and array-identifiers is really just an internal quirk of SQL Formatter implementation. We should treat both as identifiers and change the case of them.

Reply:
Array identifiers are now being converted together with ordinary identifiers - see commit [0bb7d21](https://github.com/sql-formatter-org/sql-formatter/pull/663/commits/0bb7d2137c5073a217e30f82c3fb120ce9df678b)